packages/geo/src/tms/topo50.ts

```diff
@@ -0,0 +1,183 @@
+import { Tile, TileMatrixSet, TileMatrixSetTypeOrdered } from '../tile.matrix.set.js';
+
+// Map Sheets are 24,000 x 36,000 meters, at 50_000 scale with the tile matrix set using 0.28mm / pixel
+// this calculates out to these width and height in pixels
+const tileWidth = 24_000 / (50000 * TileMatrixSet.ScaleDenominatorRatio);
+const tileHeight = 36_000 / (50000 * TileMatrixSet.ScaleDenominatorRatio);
+
+const Topo50tmsT: TileMatrixSetTypeOrdered = {
+  type: 'TileMatrixSetType',
+  title: 'Topo50 tile matrix',
+  identifier: 'Topo50',
+  orderedAxes: ['Y', 'X'],
+  boundingBox: {
+    type: 'BoundingBoxType',
+    crs: 'http://www.opengis.net/def/crs/EPSG/0/2193',
+    lowerCorner: [4578000, 2524000],
+    upperCorner: [6882000, 988000],
+  },
+  supportedCRS: 'https://www.opengis.net/def/crs/EPSG/0/2193',
+  tileMatrix: [
+    {
+      type: 'TileMatrixType',
+      identifier: '50k',
+      scaleDenominator: 50000,
+      topLeftCorner: [6882000, 988000],
+      tileWidth,
+      tileHeight,
+      matrixHeight: 64,
+      matrixWidth: 64,
+    },
+    {
+      type: 'TileMatrixType',
+      identifier: '10k',
+      scaleDenominator: 10000,
+      topLeftCorner: [6882000, 988000],
+      tileWidth,
+      tileHeight,
+      matrixHeight: 320,
+      matrixWidth: 320,
+    },
+    {
+      type: 'TileMatrixType',
+      identifier: '5k',
+      scaleDenominator: 5000,
+      topLeftCorner: [6882000, 988000],
+      tileWidth,
+      tileHeight,
+      matrixHeight: 640,
+      matrixWidth: 640,
+    },
+    {
+      type: 'TileMatrixType',
+      identifier: '1k',
+      scaleDenominator: 1000,
+      topLeftCorner: [6882000, 988000],
+      tileWidth,
+      tileHeight,
+      matrixHeight: 3200,
+      matrixWidth: 3200,
+    },
+    {
+      type: 'TileMatrixType',
+      identifier: '500m',
+      scaleDenominator: 500,
+      topLeftCorner: [6882000, 988000],
+      tileWidth,
+      tileHeight,
+      matrixHeight: 6400,
+      matrixWidth: 6400,
+    },
+  ],
+};
+
+export const Topo50Tms = new TileMatrixSet(Topo50tmsT);
+
+const SheetCodeRegex = /^[A-Z][A-Z][0-9][0-9]$/;
+const charA = 'A'.charCodeAt(0);
+
+function getTopLevelTile(tile: Tile): Tile {
+  if (tile.z === 0) return tile;
+  const pt = Topo50Tms.tileToSource({ x: tile.x + 0.5, y: tile.y + 0.5, z: tile.z });
```

Comment:
Why do we need to plus 0.5 here?

Reply:
using the center allows us to bypass some floating point issues eg x: 13.999999 vs 14.000001